tableauserverclient/models/interval_item.py

```diff
@@ -0,0 +1,166 @@
+from .property_decorators import property_is_valid_time, property_not_nullable
+
+
+class IntervalItem(object):
+    class Frequency:
+        Hourly = "Hourly"
+        Daily = "Daily"
+        Weekly = "Weekly"
+        Monthly = "Monthly"
+
+    class Occurrence:
+        Minutes = "minutes"
+        Hours = "hours"
+        WeekDay = "weekDay"
+        MonthDay = "monthDay"
+
+    class Day:
+        Sunday = "Sunday"
+        Monday = "Monday"
+        Tuesday = "Tuesday"
+        Wednesday = "Wednesday"
+        Thursday = "Thursday"
+        Friday = "Friday"
+        Saturday = "Saturday"
+        LastDay = "LastDay"
+
+
+class HourlyInterval(object):
+    def __init__(self, start_time, end_time, interval_value):
+
+        self.start_time = start_time
+        self.end_time = end_time
+        self._frequency = IntervalItem.Frequency.Hourly
+        self.interval = interval_value
+
+    @property
+    def start_time(self):
+        return self._start_time
+
+    @start_time.setter
+    @property_is_valid_time
+    @property_not_nullable
+    def start_time(self, value):
+        self._start_time = value
+
+    @property
+    def end_time(self):
+        return self._end_time
+
+    @end_time.setter
+    @property_is_valid_time
+    @property_not_nullable
+    def end_time(self, value):
+        self._end_time = value
+
+    @property
+    def interval(self):
+        return self._interval
+
+    @interval.setter
+    def interval(self, interval):
+        VALID_INTERVALS = {.25, .5, 1, 2, 4, 6, 8, 12}
+
+        if float(interval) not in VALID_INTERVALS:
+            error = "Invalid interval {} not in {}".format(interval, str(VALID_INTERVALS))
+            raise ValueError(error)
+
+        # We use fractional hours for the two minute-based intervals.
+        # Need to convert to minutes from hours here
+        if interval in {.25, .5}:
+            interval_value = int(interval * 60)
+            interval_occurrence = IntervalItem.Occurrence.Minutes
+        else:
+            interval_value = interval
+            interval_occurrence = IntervalItem.Occurrence.Hours
+
+        self._interval = [(interval_occurrence, str(interval_value))]
+
+
+class DailyInterval(object):
+    def __init__(self, start_time):
+        self.start_time = start_time
+        self._frequency = IntervalItem.Frequency.Daily
+        self.end_time = None
```

Comment:
Are you talking about the `end_time` arritbutes?

Yeah, that was so I didn't have to re-write the serializer, which I consider to be the riskiest part.

We'd have to add a bunch of `hasattr` or similar checks into the serializer, I think it's the lesser of two evils to have the empty attributes here